app/posts/finishing-touches-for-forms.md

```diff
@@ -0,0 +1,149 @@
+
+```html
+<head>
+  <%= tag.title(yield(:page_title)) %>
+  <!-- other head content -->
+</head>
+```
+
+Now we can the components library's [title with error prefix](https://govuk-components.netlify.app/helpers/title-with-error-prefix/) helper to add the 'Error:' prefix whenever `@object.errors.any?` is `true`.
+
+```ruby
+<%
+  content_for(:page_title) do
+    title_with_error_prefix(
+      "How many implements can you juggle with?",
+      error: @object.errors.any?
+    )
+  end
+%>
+```
+
+## Making sure our check boxes and radio button error links work
+
+The form builder comes with two ways to build lists of check boxes and radio buttons.
+
+The simplest is to use `#govuk_collection_check_boxes` and `#govuk_collection_radio_buttons`, which mimic the behaviour of [their Rails counterparts](https://edgeapi.rubyonrails.org/classes/ActionView/Helpers/FormBuilder.html#method-i-collection_checkboxes).
+
+You just pass in the list of options and the form builder will render them. This works for simple lists.
+
+Sometimes, however, the list needs to be customised a little further. For example we might need [a conditionally revealed questions](https://design-system.service.gov.uk/components/checkboxes#conditionally-revealing-a-related-question) or [a divider](https://design-system.service.gov.uk/components/checkboxes/#add-an-option-for-none).
+
+To support this the form builder comes with the more powerful `#govuk_check_boxes_fieldset` and `#govuk_radio_buttons_fieldset` methods which let the developer build the form field by field.
+
+This power comes at a cost. For example you could write something like this. Here the `delete` will only be shown to admins:
+
+```ruby
+f.govuk_radio_buttons_fieldset(:close_ticket)) do
+
+  if @user.admin?
+    f.govuk_radio_button(
+      :close_ticket,
+      'delete'
+    )
+  end
+
+  f.govuk_radio_button(
+    :close_ticket,
+    'archive'
+  )
+
+  # the rest of the options
+end
+```
+
+This would make it impossible for the error summary to accurately link to the first check box or radio button without repeating the logic --- an approach that's going to lead to bugs when it's updated in one place but not the other.
+
+Instead, we have to help the error summary by 'marking' the field we want the error summary to link to with `link_error: true`. This overrides the ID generation so the link in the error summary will match it.
+
+```ruby
+f.govuk_radio_buttons_fieldset(:close_ticket)) do
+
+  if @user.admin?
+    f.govuk_radio_button(
+      :close_ticket,
+      'delete',
+      link_errors: true
+    )
+  end
+
+  f.govuk_radio_button(
+    :close_ticket,
+    'archive',
+    link_errors: @user.regular_user?
+  )
+
+  # the rest of the options
+end
+```
+
+Now when there's a validation error, regardless of whether the user is an admin or not the error summary will link to the first option. It's a good idea to write some tests to ensure future changes don't affect it.
```

Comment:
Does this need a concluding paragraph? Something about how these finishing touches will make the world of difference to users or something (trite?) like that?

Reply:
The problem is we'd need a heading to make it clear the summary is for the post rather than for the third tip.

Reply:
How about a heading of "Your feedback" along the lines of:

> These finishing touches will help make sure your forms are usable and accessible.
>
> If you have any feedback or suggestions of other form features commonly missed, let us know.

Reply:
Alternative is to use an horizontal rule instead of a heading.